src/socket.cpp

```diff
@@ -182,18 +189,35 @@ void CSocket::SendPacket ( const CVector<uint8_t>& vecbySendBuf,
         // char vector in "const char*", for this we first convert the const
         // uint8_t vector in a read/write uint8_t vector and then do the cast to
         // const char*)
-        sockaddr_in UdpSocketOutAddr;
-
-        UdpSocketOutAddr.sin_family      = AF_INET;
-        UdpSocketOutAddr.sin_port        = htons ( HostAddr.iPort );
-        UdpSocketOutAddr.sin_addr.s_addr = htonl ( HostAddr.InetAddr.toIPv4Address() );
-
-        sendto ( UdpSocket,
-                 (const char*) &( (CVector<uint8_t>) vecbySendBuf )[0],
-                 iVecSizeOut,
-                 0,
-                 (sockaddr*) &UdpSocketOutAddr,
-                 sizeof ( sockaddr_in ) );
+        if (HostAddr.InetAddr.protocol() == QAbstractSocket::IPv4Protocol) {
```

Comment:
Is this necessary? `toIPv6Address()` will convert IPv4 addresses to IPv6 mapped, so I believe it should be enough to only handle the IPv6 case.